modular_skyrat/master_files/code/modules/mob/living/simple_animal/friendly/poppy.dm

```diff
@@ -63,12 +76,36 @@
 	regenerate_icons()
 
 /mob/living/simple_animal/pet/poppy/Life(delta_time = SSMOBS_DT, times_fired)
-	if(buckled || client || !DT_PROB(0.5, delta_time))
-		return ..()
+	if(buckled || client)
+		return
+
+	if(pulledby)
+		set_resting(FALSE)
+
+	if(!near_engine)
+		for(var/obj/machinery/power/supermatter_crystal/sm_crystal in view(8, src))
+			near_engine = TRUE
+			icon_state = "poppypossum_aaa"
+
+			do_jitter_animation(30)
+			manual_emote("'s fur stands up, trembling at the sight of [sm_crystal]...")
+			emote("sweatdrop")
+
+			cooldowns.Add(addtimer(CALLBACK(src, .proc/calm_down), 60 SECONDS, TIMER_STOPPABLE))
```

Comment:
Oh, there you go, now that's a good example of a callback, do that where I said to not do a sleep()